src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislike/patch/ReturnYouTubeDislikePatch.kt

```diff
@@ -137,35 +134,33 @@ class ReturnYouTubeDislikePatch : BytecodePatch(
-            context
-                .toMethodWalker(it.method)
-                .nextMethod(it.scanResult.patternScanResult!!.endIndex, true)
-                .getMethod().let { method ->
-                    with(method as MutableMethod) {
-                        // After walking, verify the found method is what's expected.
-                        if (returnType != ("Ljava/lang/CharSequence;") || parameterTypes.size != 1)
-                            return PatchResultError("Method signature did not match: $this $parameterTypes")
-
-                        val insertIndex = implementation!!.instructions.size - 1
-                        val spannedParameterRegister = instruction<OneRegisterInstruction>(insertIndex).registerA
-                        val parameter = instruction<BuilderInstruction35c>(insertIndex - 2).reference
-
-                        if (!parameter.toString().endsWith("Landroid/text/Spanned;"))
-                            return PatchResultError("Method signature parameter did not match: $parameter")
-
-                        insertShorts(insertIndex, spannedParameterRegister)
-                    }
-                }
-
-            // Additional hook, called after user dislikes.
-            with(it.mutableMethod) {
-                val insertIndex = it.scanResult.patternScanResult!!.startIndex + 2
-                val overwriteRegister = (implementation!!.instructions.elementAt(insertIndex - 1)
-                        as OneRegisterInstruction).registerA
-                insertShorts(insertIndex, overwriteRegister)
+        ShortsTextViewFingerprint.result?.let {
+            it.mutableMethod.apply {
+                val patternResult = it.scanResult.patternScanResult!!
+
+                // Boolean field that indicates if the TextView is for a dislikes button
+                val isLikesBooleanReference = instruction<ReferenceInstruction>(patternResult.endIndex).reference
+
+                val textViewFieldReference = // Like/Dislike button TextView field
+                    instruction<ReferenceInstruction>(patternResult.endIndex - 2).reference
+
+                // insert after call to parent class constructor and after a null check
+                val insertIndex = patternResult.startIndex + 6
+                addInstructions(
+                    insertIndex, """
+                    iget-boolean v0, p0, $isLikesBooleanReference  # copy boolean field
+                    if-eqz v0, :likesbutton     # if false, then this instance is for the likes button
+                    iget-object v0, p0, $textViewFieldReference   # copy TextView field
+                    invoke-static {v0}, $INTEGRATIONS_CLASS_DESCRIPTOR->updateShortsDislikes(Landroid/view/View;)Z
+                    move-result v0
+                    if-eqz v0, :likesbutton     # if zero then RYD is not enabled
+                    return-void
+                    :likesbutton                # instance is for the likes button, or RYD is not enabled 
```

Comment:
Probably this is a little more explanatory:
```suggestion
                    # Check, if the current TextView is for a dislike button
                    iget-boolean v0, p0, $isLikesBooleanReference
                    if-eqz v0, :is_like
                    
                    # Hook the TextView, if it is for the dislike button
                    iget-object v0, p0, $textViewFieldReference
                    invoke-static {v0}, $INTEGRATIONS_CLASS_DESCRIPTOR->updateShortsDislikes(Landroid/view/View;)Z
                    move-result v0
                    if-eqz v0, :ryd_disabled
                    return-void

                    :ryd_disabled
                    :is_like
```